### PR TITLE
chore(npm): update package.json and added circle config for npm publish step

### PR DESCRIPTION
**Pull request by sri-kirsh**
Reverts Pocket/apollo-cursor-pagination #4

## Goals

- add circle ci jobs to publish npm package step

### Implementation decisions:
-  circle ci config is in same format as that of [terraform-modules's circle config](https://github.com/Pocket/terraform-modules/blob/main/.circleci/config.yml). Removed terraform related configs
- in package.json, took the format of `types`, `publishConfig`, `release` and `repository` from [apollo-utils's package.json](https://github.com/Pocket/apollo-utils/blob/main/package.json)
- removed the yarn.lock file in main directory and in test directory as we don't use it in our circle ci config
- added semantic release packages to package.json
- update README

### Note:
- the npm publish steps stopped, but looks similar to the steps in terraform-module PR: https://app.circleci.com/pipelines/github/Pocket/terraform-modules/2085/workflows/5de196e9-7fb6-4c5b-8861-f356d5456b08/jobs/1885
- my guess is it might get triggered after a merge to main.
-  able to run test with new commands but it fails (also fails in main with yarn commands). 

JIRA Ticket: [INFRA-213 ](https://getpocket.atlassian.net/browse/INFRA-213)

old PR is here: https://github.com/Pocket/apollo-cursor-pagination/pull/3

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
+  "name": "@pocket-tools/apollo-cursor-pagination",
+  "version": "1.0.0",
+  "description": "Relay's Connection implementation for Apollo Server GraphQL library with primary key support for cursor generation.",
+  "keywords": [
+    "apollo",
+    "pagination"
+  ],
   "main": "dist/index.js",
-  "repository": "https://github.com/Terminal-Systems/apollo-cursor-pagination",
+  "types": "dist/index.d.ts",
```
Comment (kschelonka) [inline package.json]: We have no types

**Comment by kschelonka** [comment]
When this was private and just used by list-api we didn't update the readme, but I don't think those installation instructions are correct anymore.

File: package.json (hunk 1 above)
Comment (sri-kirsh) [reply package.json]: could be old diff.. it seems like removed and i dont see it in the package.json,

Comment (kschelonka) [reply package.json]: Maybe you didn't push local changes? I'm still seeing it on this PR after refresh.

Comment (sri-kirsh) [reply package.json]: lol now it shows as updated even if the only change was with README file

Comment (kschelonka) [reply package.json]: are you seeing it now?

**Comment by sri-kirsh** [comment]
> When this was private and just used by list-api we didn't update the readme, but I don't think those installation instructions are correct anymore.

update readme installation instruction, description was earlier updated...everything below looks like how we use the pagination, and not sure about the test...do we have to update these `yarn` to `npm`?

File: package.json (hunk 1 above)
Comment (sri-kirsh) [reply package.json]: no i am not..it shows as `outdated`

Comment (sri-kirsh) [reply package.json]: oh i see, the `types` got removed in this commit: https://github.com/Pocket/apollo-cursor-pagination/pull/5/commits/5b553f9e4d93ed8700f5648aa32c3db25fafd1a2 but somehow its not showing up in your UI, confused why

Comment (kschelonka) [reply package.json]: it's all good now!

**Comment by kschelonka** [comment]
> update readme installation instruction, description was earlier updated...everything below looks like how we use the pagination, and not sure about the test...do we have to update these yarn to npm?

I think so, but double check by getting everything running locally.

**Review by kschelonka — APPROVED** [review]

**Pull request merged**